flume-ng-sdk/src/main/java/org/apache/flume/api/AbstractRpcClient.java

```diff
@@ -61,4 +64,26 @@ public abstract void appendBatch(List<Event> events)
   protected abstract void configure(Properties properties)
       throws FlumeException;
 
+  public static int parseBatchSize(Properties properties) {
```

Comment:
AbstractRpcClient does not seem to be best place for this static method.
_batchSize looks a bit strange in java.
AbstractRpcClient.parseBatchSize(clientProps) call in AbstractRpcSink too.

I would consider to move the method to a utility class (or maybe to RpcClient).

Reply:
I see your points, but here is some explanation:
AbstractRpcClient.parseBatchSize(clientProps) call exists in AbstractRpcSink because at the time of the config check it does not have a configured client. Otherwise the client handles batch sizes. I will put that in as a comment in the code. 
So the parse belongs to the client as it is their configuration. This is not a generic utility that is why I did not put in a separate class. This part of the flume uses a "batch-size" in the config while other parts use "batchSize".

Reply:
I see, thanks for the explanation.
I did not see that AbstractRpcSink uses different batch size at configuration / processing time (same value, but coming from different places).
The comment wil be fine.